goal_src/engine/game/generic-obs.gc

```diff
@@ -144,61 +229,485 @@
+     )
     )
    )
-  )
-
-(defbehavior process-release? process ((arg0 process))
-  (let ((gp-0 (command-get-process arg0 *target*)))
-   (the-as
-    symbol
-    (if (and gp-0 (let ((a1-1 (new 'stack-no-clear 'event-message-block)))
-                   (set! (-> a1-1 from) self)
-                   (set! (-> a1-1 num-params) 1)
-                   (set! (-> a1-1 message) 'query)
-                   (set! (-> a1-1 param 0) (the-as uint 'mode))
-                   (send-event-function gp-0 a1-1)
-                   )
-         'target-grab
-         )
-     (send-event gp-0 'end-mode)
-     #t
+  :trans
+  (behavior ()
+   (if (!= (-> self cur-trans-hook) (-> self new-trans-hook))
+    (set! (-> self cur-trans-hook) (-> self new-trans-hook))
+    )
+   (if (!= (-> self cur-post-hook) (-> self new-post-hook))
+    (set! (-> self cur-post-hook) (-> self new-post-hook))
+    )
+   (when
+    (and
+     (-> self new-joint-anim)
+     (nonzero? (-> self skel))
+     (and (!= (if (> (-> self skel active-channels) 0)
+               (-> self skel root-channel 0 frame-group)
+               )
+           (-> self new-joint-anim)
+           )
+      (!= (-> self anim-mode) 'clone-anim)
+      )
+     )
+    (ja-channel-push! 1 (the-as int (-> self new-joint-anim-blend)))
+    (let ((gp-0 (-> self skel root-channel 0)))
+     (joint-control-channel-group-eval!
+      gp-0
+      (-> self new-joint-anim)
+      num-func-identity
+      )
+     (set! (-> gp-0 frame-num) 0.0)
      )
     )
-   )
-  )
-
-(define-extern camera-change-to (function string int symbol none)) ;; TODO - not confirmed yet
-;; TODO - for projectiles | dark-eco-pool
-(define-extern part-tracker-init (function sparticle-launch-group int basic basic basic vector none)) ;; TODO - not confirmed
-;; TODO - for target-part
-(define-extern process-drawable-random-point! (function process-drawable vector vector))
-
-
-;; TODO PLACEHOLDER STATE
-(defstate manipy-idle (manipy)
-  :code (behavior ()
-    (loop
-      (suspend)
+   (let ((v1-20 (handle->process (-> self cur-grab-handle))))
+    (when v1-20
+     (let ((gp-1 (-> (the-as process-drawable v1-20) root trans)))
+      (cond
+       ((type-type? (-> self root type) collide-shape)
+        (let* ((a0-13 (-> self root))
+               (t9-3
+                (method-of-object (the-as collide-shape a0-13) TODO-RENAME-28)
+                )
+               )
+         (vector-! (new 'stack-no-clear 'vector) gp-1 (-> self old-grab-pos))
+         (t9-3 (the-as collide-shape a0-13))
+         )
+        )
+       (else
+        (vector+!
+         (-> self root trans)
+         (-> self root trans)
+         (vector-! (new 'stack-no-clear 'vector) gp-1 (-> self old-grab-pos))
+         )
+        )
+       )
+      (set! (-> self old-grab-pos quad) (-> gp-1 quad))
       )
+     )
     )
+   ((-> self cur-trans-hook))
+   (none)
+   )
+  :code
+  (behavior () (none))
+  ;  (logclear! (-> self mask) (process-mask heap-shrunk))
```

Comment:
FYI we need actually functional manipys, make this body just `(loop (suspend))` instead of `(none)`